### DISCUSSION TITLE
UIHook for Integrations removed?

### DISCUSSION THREAD

**Discussion by lukeshumard** [vercel/vercel#5903] [Help] [answered]
I've been trying to create an integration based off [this Datadog logs example](https://github.com/vercel/integrations/tree/master/datadog-logs) in the [Integrations repository](https://github.com/vercel/integrations).

I have gotten my Integration created in the Integrations Console, but I only see a URL for **Redirect URL**. Once I've installed the Integration, I am presented with a screen with the message...
> {YOUR_INTEGRATION} does not provide custom UI.

I've found older / outdated documentation online that describes an additional input for the **UIHook URL** in the Integrations console, but I cannot find any mention of this in any current documentation or the Integrations guide. Has the UIHook been deprecated? If so, is there somewhere I could find documentation on how to build an Integration with a custom UI?

**Comment by chriswdmr** [comment, marked as answer]
Hey there, thank you for reaching out!

Indeed, we removed/deprecated the UI Hooks last year.
We're in the process of cleaning up those outdated examples as well as providing a better documentation for integrations in general. 

For now, we're only supporting integrations with the callback flow.
This means the user will see a popup with your defined Redirect URL.
Inside this popup you can provide your own UI and redirect the user back once you're done with the setup.

You can take a look at this example https://github.com/vercel/example-integration. It's a very early version but this shows a working example including all steps a user will take during an installation.

Hope this helps!

**Reply by lukeshumard** [reply]
Thanks for the clarity and integration example. I appreciate the help.
